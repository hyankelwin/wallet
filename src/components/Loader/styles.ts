import styled, { keyframes } from 'styled-components';
import Color from '../../styles/color';

interface ListItemProps {
  size?: string;
}

export const Container = styled.div<ListItemProps>`
  display: flex;
  flex-direction: column;
  align-items: center;

  p {
    font-size: 16px;
    text-align: center;
    color: #666360;
    margin-top: 20px;
  }
`;

const SpinnerAnimation = keyframes`
  0% { transform: rotate(0deg); }
  100% { transform: rotate(360deg); }
`;

export const Spinner = styled.div<ListItemProps>`
  font-size: 0.5rem;
  position: relative;
  text-indent: -9999em;
  border-radius: 50%;
  width: ${props => props.size || '3rem'};
  height: ${props => props.size || '3rem'};
  border-top: 0.3rem solid rgba(255, 255, 255, 0.8);
  border-right: 0.3rem solid rgba(255, 255, 255, 0.8);
  border-bottom: 0.3rem solid rgba(255, 255, 255, 0.8);
  border-left: 0.3rem solid ${Color.primary.default};
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
  -webkit-animation: ${SpinnerAnimation} 0.6s infinite linear;
  animation: ${SpinnerAnimation} 0.6s infinite linear;

  &:after {
    border-radius: 50%;
    width: 3rem;
    height: 3rem;
  }
`;
