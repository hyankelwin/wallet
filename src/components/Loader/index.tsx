import React from 'react';
import { Container, Spinner } from './styles';

interface LoaderProps {
  text?: string;
  size?: string;
}

const Loader: React.FC<LoaderProps> = ({ text, size }) => (
  <Container>
    <Spinner size={size} />
    {!!text && <p>{text}</p>}
  </Container>
);

export default Loader;
