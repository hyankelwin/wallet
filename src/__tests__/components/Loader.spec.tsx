import { mount } from 'enzyme';
import Loader from '../../components/Loader';

describe('Loader component', () => {
  it('should be able to render Loader component with success', () => {
    const wrapper = mount(<Loader text="Carregando" />);

    expect(wrapper.debug()).toMatchSnapshot();
  });
});
