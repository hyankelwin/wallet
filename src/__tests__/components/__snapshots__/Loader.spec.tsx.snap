// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader component should be able to render Loader component with success 1`] = `
"<Loader text=\\"Carregando\\">
  <styled.div>
    <div className=\\"sc-bdvvtL jPiZSA\\">
      <styled.div size={[undefined]}>
        <div size={[undefined]} className=\\"sc-gsDKAQ jJnrhU\\" />
      </styled.div>
      <p>
        Carregando
      </p>
    </div>
  </styled.div>
</Loader>"
`;
